config/initializers/paperclip.rb

```diff
@@ -0,0 +1,17 @@
+Paperclip::Attachment.default_options.merge!(
+  path: ':rails_root/uploads/:class/:attachment/:id_partition/:style/:filename',
+  url: '/uploads/:class/:id/:style'
```

Comment:
Let's wait till we get to that bridge before we cross it.

On Thu, Jan 21, 2016 at 2:53 PM, Melody <notifications@github.com> wrote:

> In config/initializers/paperclip.rb
> <https://github.com/thecartercenter/elmo/pull/175#discussion_r50453592>:
>
> > @@ -0,0 +1,17 @@
> > +Paperclip::Attachment.default_options.merge!(
> > +  path: ':rails_root/uploads/:class/:attachment/:id_partition/:style/:filename',
> > +  url: '/uploads/:class/:id/:style'
>
> i'm not entirely sure how their interpolation works under the hood, but
> i'm not 100% sure if that's possible. I think we may have to denormalize
> the mission slug onto the media model if we're going to make that work with
> acceptable performance, and I'd have to figure out what we can do about the
> locale. You're not actually defining like, a proper rails routing string
> there, so I think all of the :blah things in the string need to be
> methods on the model with the attachment. We can try a few things, and I
> can dig deeper into their docs on the related issue, but it could be
> difficult.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thecartercenter/elmo/pull/175/files#r50453592>.
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org